affirm/src/main/java/com/affirm/android/CheckoutRequest.java

```diff
@@ -98,10 +98,13 @@ public void create() {
         metadataJson.addProperty(PLATFORM_AFFIRM_KEY, PLATFORM_AFFIRM_VALUE);
 
         final JsonObject checkoutJson = jsonParser.parse(gson.toJson(checkout)).getAsJsonObject();
-        JsonObject address = jsonParser
-                .parse(gson.toJson(Objects.requireNonNull(checkout.shippingAddress()).address()))
-                .getAsJsonObject();
-        ((JsonObject) checkoutJson.get(SHIPPING)).add(ADDRESS, address);
+        final Shipping shipping = checkout.shippingAddress();
+        if (shipping != null) {
+            JsonObject address = jsonParser
+                    .parse(gson.toJson(shipping.address()))
+                    .getAsJsonObject();
+            ((JsonObject) checkoutJson.get(SHIPPING)).add(ADDRESS, address);
+        }
         checkoutJson.add(MERCHANT, merchantJson);
```

Comment:
There is a lot of repeated code here. You should make some utilities to parse data and/or build your final json.

Reply:
@kevindavmoore It has been modified.

Reply:
Looks much better